ui/apps/platform/src/routePaths.ts

```diff
@@ -102,6 +106,119 @@ export const vulnerabilitiesWorkloadCveDeploymentSinglePath = `${vulnerabilities
 export const vulnerabilityReportsPath = `${vulnerabilitiesBasePath}/reports`;
 export const vulnerabilityReportPath = `${vulnerabilitiesBasePath}/reports/:reportId`;
 
+// Source of truth for conditional rendering of Body route paths and NavigationSidebar links.
+
+type RouteDescription = {
+    featureFlagDependency?: FeatureFlagEnvVar; // can add array as alternative if needed in the future
+    resourceAccessRequirements: ResourceName[]; // assume READ_ACCESS
```

Comment:
I know READ_ACCESS is the only use case today, but thinking of possible future requirements, is it worth splitting this into separate checks for READ and READ_WRITE levels?

Reply:
To lower barrier to entry, and because it seems straightforward how to extend, my gut feeling is to defer.